### PR TITLE
Vpn always on & kill switch android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/37013
Resolves https://github.com/brave/brave-browser/issues/23193

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/AndroidManifest.xml [hunk 1]
```diff
@@ -96,6 +96,14 @@
     android:screenOrientation="sensorPortrait"
     tools:ignore="LockedOrientationActivity,DiscouragedApi"/>
 
+<activity
+    android:name="org.chromium.chrome.browser.vpn.activities.VpnAlwaysOnActivity"
+    android:theme="@style/Theme.Chromium.Activity"
+    android:noHistory="true"
+    android:launchMode="singleTask"
+    android:screenOrientation="sensorPortrait"
+    tools:ignore="LockedOrientationActivity,DiscouragedApi"/>
```
Comment (SergeyZhukovsky) [inline android/java/AndroidManifest.xml:105]: what API in particular triggers the `DiscouragedApi` warning?

File: android/java/res/values-night/brave_colors.xml [hunk 2]
```diff
@@ -251,4 +251,8 @@
     <color name="vpn_timer_bg_color">#0D0F14</color>
     <color name="vpn_timer_text_color">#F6F7F8</color>
     <color name="vpn_timer_icon_color">#A1ABBA</color>
+
+    <color name="text_primary">#F3F5F7</color>
```
Comment (SergeyZhukovsky) [inline android/java/res/values-night/brave_colors.xml:255]: maybe a comment of what colours for or name of colours different like others we have for vpn `vpn...`

File: browser/ui/android/strings/android_brave_strings.grd [hunk 3]
```diff
@@ -3932,6 +3932,30 @@ If you don't accept this request, VPN will not reconnect and your internet conne
...
+      How to turn on:
+    </message>
+    <message name="IDS_KILL_SWITCH_TUTORIAL_TEXT_1" desc="Vpn kill swicth tutorial text">
+      1. Open your device's settings from your app drawer or tapping the button below.
+    </message>
+    <message name="IDS_KILL_SWITCH_TUTORIAL_TEXT_2" desc="Vpn kill swicth tutorial text">
+      2. Find Brave VPN and tap on the settings icon to the right.
+    </message>
+    <message name="IDS_KILL_SWITCH_TUTORIAL_TEXT_3" desc="Vpn kill swicth tutorial text">
+      3. Toggle <ph name="ALWAYS_ON_TUTORIAL_START">&lt;always_on_tutorial&gt;</ph>Always-on VPN<ph name="ALWAYS_ON_TUTORIAL_END">&lt;/always_on_tutorial&gt;</ph> to enable it.\n4. Toggle <ph name="ALWAYS_ON_TUTORIAL_2_START">&lt;always_on_tutorial_2&gt;</ph>Block connections without VPN<ph name="ALWAYS_ON_TUTORIAL_2_END">&lt;/always_on_tutorial_2&gt;</ph> to enable the kill switch.
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:3951]: I would do 1, 2, 3, 4 rather or together or 3 and 4 as separate strings as 1, 2.

File: android/java/res/layout/activity_vpn_always_on.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            style="@style/DefaultRegular"
+            android:textColor="@color/text_secondary"
+            android:text="@string/kill_switch_text"
+            android:gravity="center_horizontal"/>
+    </LinearLayout>
+</ScrollView>
```
Comment (tapanmodh) [inline android/java/res/layout/activity_vpn_always_on.xml:66]: nit: empty line

**Review by tapanmodh — APPROVED** [review]

File: android/java/res/values-night/brave_colors.xml (hunk 2 above)
Comment (deeppandya) [reply android/java/res/values-night/brave_colors.xml:255]: i am using the names from the nala tokens. and moving forward i am planning to replace existing colors with nala tokens to keep it uniform across different platforms

File: android/java/AndroidManifest.xml (hunk 1 above)
Comment (deeppandya) [reply android/java/AndroidManifest.xml:105]: @SergeyZhukovsky it's mainly because of the portrait lock here `android:screenOrientation="sensorPortrait"`

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**